src/main/java/org/opentripplanner/service/vehiclerental/street/VehicleRentalEdge.java

```diff
@@ -206,4 +211,11 @@ private static Set<RentalFormFactor> allowedModes(StreetMode streetMode) {
       default -> Set.of();
     };
   }
+
+  public Optional<Double> getCurrentRangeMeters() {
+    if (vertex.getStation() instanceof VehicleRentalVehicle) {
+      return ((VehicleRentalVehicle) vertex.getStation()).getCurrentRangeMeters();
+    }
+    return Optional.empty();
```

Comment:
Please move this into the domain model, ie. have a method on VehicleRentalPlace that encapsulates this.

Reply:
added it to VehicleRentalPlace and were able to remove the if-check as we have implementations in each of the specific Edge-Types